Engine/TransactionHandlers/BrokerageTransactionHandler.cs

```diff
@@ -820,11 +818,16 @@ private void HandleOrderEvent(OrderEvent fill)
             {
                 Log.Debug("BrokerageTransactionHandler.HandleOrderEvent(): " + fill);
                 Interlocked.Exchange(ref _lastFillTimeTicks, DateTime.Now.Ticks);
+
+                // set the price currency on the fill and on the order
+                var security = _algorithm.Securities[fill.Symbol];
+                fill.FillPriceCurrency = security.SymbolProperties.QuoteCurrency;
+                order.PriceCurrency = fill.FillPriceCurrency;
```

Comment:
I agree with there being lots of changes if we were to update the brokerages to do this. Maybe the best way to enforce this is by making the currency a ctor parameter on the OrderEvent class. This will minimally force the concern everywhere it is constructed.

I am very hesitant to check if it's not set and then set it. I'd prefer us go one way or the other, either we always set it here and it is assumed that the brokerage populates the fills using the correct currency as specified on the security object, or the brokerage populates the order event and the transaction handler merely passes the data through the system. My preference here would be the latter since that's how things are currently handled. In most cases the currency should be easily available. For example, tradier will always produce USD denominated fills. I believe IB transmits the currency information via the contract details. Oanda/Fxcm I'm not as familiar with, but could use the `ISecurityProvider` to get access to the correct currency for the security.  I think the bigger question is what do we do if we encounter a mismatch, for example, a brokerage impl sends a fill that's quoted in currency A, but the security is configured to use currency B... that I do not currency have an answer for :)